### PR TITLE
Add the ability to add extra env variables

### PR DESCRIPTION
**Pull request by zpetterd**
**What this PR does / why we need it**:
This affects the `incubator/fluentd-cloudwatch` chart.

This adds the ability for the user to add custom environment variables to daemonset pod. For example the below could be added
```
- name: NODE_NAME
   valueFrom:
     fieldRef:
       fieldPath: spec.nodeName
```
Which allows the result to be used like
```
<filter kube-api-server-audit.*>
  @type record_modifier

  <record>
    node_hostname "#{ENV['NODE_NAME']}"
  </record>
</filter>
```
in the Fluentd config like

@jmcarp , @icereval , @kenden

### CODE REVIEW

File: incubator/fluentd-cloudwatch/README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ The following tables lists the configurable parameters of the Fluentd Cloudwatch
 | `rbac.create`                   | If true, create & use RBAC resources       | `false`                                                    |
 | `rbac.serviceAccountName`       | existing ServiceAccount to use (ignored if rbac.create=true) | `default`                                |
 | `tolerations`                   | Add tolerations                            | `[]`                                                       |
+| `envs`                          | Add pod environment variables (must be specified as a single line object)             | `[]`
```
Comment (unguiculus) [inline incubator/fluentd-cloudwatch/README.md]: Please rename this to `extraVars`. There are already quite some charts to use this. Let's be consistent.

File: incubator/fluentd-cloudwatch/test-values.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+envs:
```
Comment (unguiculus) [inline incubator/fluentd-cloudwatch/test-values.yml]: What is this file for?

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by zpetterd** [comment]
@unguiculus I have made those changes

**Review by unguiculus — APPROVED** [review]

**Pull request merged**